rpcserver.go

```diff
@@ -1355,6 +1355,197 @@ func (r *rpcServer) SendPaymentSync(ctx context.Context,
 	}, nil
 }
 
+// SendToRoute dispatches a bi-directional streaming RPC for sending payments
```

Comment:
Probably that shouldn't be done in this `PR`, but from my POV this diff can be significantly reduced, how we might achieve this in the future?
- Current `routing.SendPayment` basically consist of three logic blocks:
  - check the route cache
  - find the route
  - send payment to route

If we move work with cache in the `routing.FindRoute` function, we basically could remove the `routing.SendPayment` function and instead operate with `routing.FindRoute` inside the rpc server.

With updates above the function which I mentioned in the previous comment might use the core `sendPayment` function. Probably I will do that in the future updates, with `Laolu` approval.